src/ngAria/aria.js

```diff
@@ -310,6 +320,9 @@ ngAriaModule.directive('ngShow', ['$aria', function($aria) {
             return true;
           }
         }
+        if (!elem.attr('role') && !isNodeOneOf(elem, ['BUTTON', 'A'])) {
```

Comment:
Great point. I updated the blacklist and added it to the dynamic keypress binding so there would be more consistency.